python/tutorials/06-fused-attention.py

```diff
@@ -95,7 +95,7 @@ def keep(conf):
     return True
 
 
-@triton.autotune(list(filter(keep, configs)), key=["N_CTX"])
+@triton.autotune(list(filter(keep, configs)), key=["N_CTX", "HEAD_DIM"])
```

Comment:
When you add an arg in the key, you need to make sure the arg precedes `BLOCK_M` and `BLOCK_N` in the function arg list. Otherwise, there will be this "list index out of range" error as in the CI.
I can also reproduce the error on A100. The NV CI is not complaining because it's running the tutorial as an interpreter test.
More details about the error: the indices of the keys are assigned from the arg list including `BLOCK_M` and `BLOCK_N`. However, they are later accessed from a list that does not include `BLOCK_M` and `BLOCK_N`. I'm not sure if this is a bug that needs to be fix. But at least we know why the CI is failing.

Reply:
Thanks @zhanglx13! My original case has 3 different head_dims: 64, 128, 256, when running it, tuning only happened for the first head_dim, so we get suboptimal perf for head_dims of 128 and 256. I tried adding H to the key, it starts to retune for the last two cases. But adding HEAD_DIM to the key doesn't work (no retune for the last two cases).

I will try to see if I can repro on OSS and come up with a proper fix.